www/scipylib/building/windows.rst

```diff
@@ -32,185 +32,231 @@ information than this document will provide.
-
-The other option is to create a configuration file, either called ``site.cfg`` or ``.numpy-site.cfg``. If you are building
-both NumPy and SciPy, you should store it in your ``C:\Users\{username}`` directory of your native Windows environment or
-your ``$HOME`` or ``~`` directory if you are using Cygwin_. If you are just building NumPy, you can store it in the
-same directory as the topmost ``setup.py`` file. Before filling it in, make sure that your configuration file can be detected by
-filling it with some invalid text (e.g. "asdf") and then run ``python setup.py config`` again. An exception should be thrown
-because Python won't be able to parse your configuration file.
-
-Depending on which library you use, the exact specifics of the configuration file will vary. The ``site.cfg.example``
-file, which should be located at the top of your NumPy installation, provides an excellent guide for how to fill in
-your configuration file given the libraries you are using. If you do not have such a file, you can find it online `here <https://github.com/numpy/numpy/blob/master/site.cfg.example>`__.
+
+Building the Released SciPy
+---------------------------
+
+This section provides the step-by-step process to build the released scipy. If you want
+to build completely from source, you should estimate at least three hours to build all
+libraries and compile SciPy. Feel free to stop and inspect any step at any time, but
+for this section, we'll just mention the steps without providing an in-depth explanation
+for the reasons behind them. If you have further questions about what we're doing, more
+in-depth documentation is provided in the sections below. Also, please make sure to read
+this section before proceeding, as the presence or absence of error messages in general
+is not a good indication of whether you've completed a step correctly. Each step creates
+particular files, and what ultimately matters is whether you have built the required files
+rather than whether error messages appeared in your terminal.
+
+Building OpenBLAS
+=================
+
+First, we need to install the software required to build OpenBLAS_, which is the BLAS_
+library that we're going to use. Because the software to build OpenBLAS is different than
+that required to build SciPy and because OpenBLAS takes a long time to build, we're going
+to start building OpenBLAS first and then explain what to do next while the OpenBLAS build
+is running. **Alternatively, if you'd rather download a pre-build OpenBLAS, download the**
+**one of the** `pre-built zip files`_ **and skip to the Installing OpenBLAS section below.**
+
+First, install MSYS2 using `these instructions`_. Make sure to install the correct
+architecture for the SciPy that you want to build (eg. 32 or 64 bit), and make sure to
+install the MinGW, and git packages. Open a MSYS2 terminal and type the following:
+
+.. code:: shell
+
+   pacman -S --needed base-devel 
```

Comment:
I think splitting these for 32bit and 64 bit is better for one-liners. So I would propose:

```
First, install MSYS2 using `these instructions`_ including the `pacman` update instructions.
Occasionally during the updates the terminal might ask you to close the terminal but 
refuses to be closed and hang. If this happens you can kill it via Task Manager and continue 
with the instructions. Make sure to install the correct architecture for the SciPy that you 
want to build (eg. 32 or 64 bit). Now, you have three options for opening a terminal 
which are MSYS2, MINGW (32bit,64bit). After updating all the packages, now we are 
ready to install some more package bundles that we will need. Open a MSYS2 terminal 
and type the following depending on the architecture of your choice:

.. code:: shell

    pacman -S --needed base-devel mingw-w64-i686-toolchain mingw-w64-i686-cmake git

for 32-bit and 

.. code:: shell

    pacman -S --needed base-devel mingw-w64-x86_64-toolchain mingw-w64-x86_64-cmake git

for 64-bit build. It will prompt to whether install everything in these packages and 
you can simply accept all via hitting enter key at each step. 

We should be aware of the fact that these tools also install Python2, very similar to 
a virtual environment, which is only usable within MSYS2 terminals and we are 
**not** going to use it at any point. Therefore, after the OpenBLAS build we are 
done with MSYS2 and its terminals. We will not use for any step for building SciPy. 
Please keep this in mind to avoid nasty conflicts. 

You should now have all of the tools required to build OpenBLAS. Now change directories
to your favorite location with the following:
```